service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/disk/ControlledAzureDiskResource.java

```diff
@@ -73,9 +74,23 @@ public ControlledAzureDiskResource(DbResource dbResource) {
     validate();
   }
 
+  public static ControlledAzureDiskResource.Builder builder() {
+    return new ControlledAzureDiskResource.Builder();
+  }
+
   /** {@inheritDoc} */
   @Override
-  public Optional<UniquenessCheckAttributes> getUniquenessCheckParameters() {
+  @SuppressWarnings("unchecked")
+  public <T> T castByEnum(WsmResourceType expectedType) {
```

Comment:
I'd name this `toResourceType()` instead. The fact that it's a cast is an implementation detail.